src/net.h

```diff
@@ -874,6 +884,26 @@ class CNode
         assert(false);
     }
 
+    std::string ConnectionTypeToString() const
```

Comment:
I'm not sure it's better to have `ConnectionTypeToString` as a method on CNode instead of a free function that simply takes a `ConnectionType`.

But if so, `ConnectionTypeAsString` might be a better name.

Reply:
hm, I got a lot of review on #19316 expressing concerns around publicly exposing `m_conn_type`, which led to the current design of wrapper classes. since I need the return value in `net_processing`, I've opted to keep the function on `CNode` & updated the name to better reflect behavior, as per your suggestion. happy to reconsider if you still feel it would be better as a free function. 

also moved into `net.cpp`. I'm still trying to understand when inlining is worth it. so far my heuristic has mainly been "is this function computationally very simple". but I hear your point of header files being readable. I'm interested in learning more if you have any other heuristics you consider / resources to share.